### PR TITLE
Feature/116 store timetable locally

### PR DESCRIPTION
**Pull request by lukefisklennon**
Stores the selected courses and timetable data locally

Edit: oops forgot to request reviews

### CODE REVIEW

**Comment by maxwowo** [comment]
I'm getting this error screen after selecting a course, closing the tab, then reopening `localhost:3000`
image

**Comment by lukefisklennon** [comment]
That's so strange, I was getting that issue too, then it went away for a while, now it's back 🤔 I'll look into it now

**Comment by maxwowo** [comment]
> That's so strange, I was getting that issue too, then it went away for a while, now it's back 🤔 I'll look into it now

Oof that's tough to debug, below is exactly what I did, maybe it'll help
1. npm run start:mock
2. Choose COMP9243
3. Close tab
4. Reopen localhost:3000

**Comment by lukefisklennon** [comment]
> Oof that's tough to debug, below is exactly what I did, maybe it'll help

Ah thanks, as it turns out the bug would happen if there was high enough network latency, so it appeared to happen at random. Should be fixed now.

**Review by maxwowo — CHANGES_REQUESTED** [review]
Just a minor remark, everything else looks good

File: client/src/App.tsx [hunk 1]
```diff
@@ -62,6 +69,19 @@ const Footer = styled(Box)`
...
+  const isInitialMount = React.useRef(true);
+
+  useEffect(() => {
+    if (isInitialMount.current) {
+      isInitialMount.current = false;
+    } else {
+      effect();
+    }
+  }, dependencies);
+};
```
[the remark this answers is not in the document]
Comment (lukefisklennon) [reply client/src/App.tsx:83]: Ah yes, good idea

**Review by maxwowo — APPROVED** [review]

**Review by esyw — APPROVED** [review]

**Pull request merged**